libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraComponent.kt

```diff
@@ -1,147 +1,177 @@
-            // no op
-        }
-    }
+    private lateinit var gesturesHandler: NavigationBasicGesturesHandler
 
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
-        viewportDataSource = MapboxNavigationViewportDataSource(mapView.getMapboxMap())
-        navigationCamera = NavigationCamera(
-            mapView.getMapboxMap(),
-            mapView.camera,
-            viewportDataSource
-        )
-        mapView.gestures.addOnMoveListener(onMoveListener)
+        gesturesHandler = NavigationBasicGesturesHandler(navigationCamera)
+        mapView.camera.addCameraAnimationsLifecycleListener(gesturesHandler)
+
         coroutineScope.launch {
-            cameraViewModel.state.collect {
-                when (it.cameraAnimation) {
-                    is CameraAnimate.EaseTo -> {
-                        mapView.camera.easeTo(it.cameraOptions)
-                    }
-                    is CameraAnimate.FlyTo -> {
-                        mapView.camera.flyTo(it.cameraOptions)
+            dropInNavigationViewContext.viewModel.navigationState.collect {
+                when (it) {
+                    is NavigationState.FreeDrive -> {
+                        viewportDataSource.followingZoomPropertyOverride(
+                            FOLLOWING_ZOOM_OVERRIDE_FREE_DRIVE
+                        )
+                        viewportDataSource.overviewZoomPropertyOverride(
+                            OVERVIEW_ZOOM_OVERRIDE_FREE_DRIVE
+                        )
                     }
-                    is CameraAnimate.SetTo -> {
-                        mapView.getMapboxMap().setCamera(it.cameraOptions)
+                    else -> {
+                        viewportDataSource.clearFollowingOverrides()
+                        viewportDataSource.clearOverviewOverrides()
                     }
                 }
-                it.location?.let { enhancedLocation ->
-                    updateCamera(it.cameraMode, enhancedLocation)
+            }
+        }
+
+        coroutineScope.launch {
+            cameraViewModel.state.collect { state ->
+                if (state.isCameraInitialized) {
+                    requestCameraModeTo(cameraMode = state.cameraMode)
                 }
-                when (it.cameraTransition) {
-                    is CameraTransition.ToFollowing -> {
-                        navigationCamera.requestNavigationCameraToFollowing()
-                    }
-                    is CameraTransition.ToOverview -> {
-                        navigationCamera.requestNavigationCameraToOverview()
-                    }
-                    is CameraTransition.ToIdle -> {
-                        navigationCamera.requestNavigationCameraToIdle()
+                viewportDataSource.overviewPadding = overviewEdgeInsets
+                viewportDataSource.followingPadding = followingEdgeInsets
+                viewportDataSource.evaluate()
```

Comment:
nit: this can be set only once on creation since the values are static anyway, at least for now